[Guide] ES6: Prevent object mutation; fixes and enhancements

AdrianSkar opened the pull request:
- Remove phantom image
- Remove notes for contributors
- Change explanation and hints.
- Links to markdown

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

AdrianSkar committed:
[Guide] ES6: Prevent object mutation; fixes and enhancements

- Remove phantom image
- Remove notes for contributors
- Change explanation and hints.
- Links to markdown

RandellDawson reviewed (changes requested):
Please capitalize all words in headers except articles like `a`, `the`, etc.

AdrianSkar committed:
Changed as requested

Changed titles as requested by @randelldawson. 
Note: I'm just used to [MDN](https://developer.mozilla.org/en-US/docs/MDN/Contribute/Guidelines/Writing_style_guide#Page_titles) and other language styles and though it was clearer. Is it ok now for this guide?

RandellDawson committed:
fix: removed repl.it link

RandellDawson commented:
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉
